src/Livewire/Order/Order.php

```diff
@@ -526,9 +528,23 @@ public function fetchContactData(bool $replicate = false): void
     }
 
     #[Renderless]
-    public function updatedOrderState(): void
+    public function saveStates(): void
     {
-        $this->getAvailableStates('state');
+        try {
+            UpdateOrder::make([
+                'id' => $this->order->id,
+                'state' => $this->order->state,
+                'payment_state' => $this->order->payment_state,
+                'delivery_state' => $this->order->delivery_state,
+            ])
+                ->checkPermission()
+                ->validate()
+                ->execute();
+
+            $this->getAvailableStates('state');
```

Comment:
shouldn't the available states for delivery_state (and maybe payment_state) also be updated?